### ISSUE TITLE
web3-eth-abi is not handling contract responses properly when the result is `0x`

### ISSUE THREAD

**Issue by ianaya89** [ethereum/web3.js#1639]
I am trying to interact with some contract methods after creating the contract instance (new web3.eth.Contract).
Some of the contract methods will return `0x` depending on the scenario and new web3 version (`1.0.*`) is not handling that properly. I tested that using the stable version and everything works fine.

This is the `eth_call` we are doing through JSON RPC:
```
POST / {"jsonrpc":"2.0","id":1,"method":"eth_call","params":[{"data":"0x3defb962","to":"0x61b0219e056b6d6c67f4aeb6cd5f6bf74a7aac50"},"latest"]}

RESPONSE / {"jsonrpc":"2.0","id":1,"result":"0x"}
```

`v0.20.*` is handling that properly and the result of the method call is parsed as `0`, but in `v1.0.*` we are getting this error message: `Couldn't decode bytes8 from ABI: 0x The size doesn't match`.

After digging in the code, I found that `decodeParameter` method (at `web3-eth-abi/src/index.js`) is checking if `decodedValue === '0x'` but that will never happen because in the previous line, the string `0x` is being replaced by an empty string. So 

```javascript
outputs.forEach(function (output, i) {
  // This line throws the error because is not possible to decoded the empty string
  var decodedValue = solidityTypes[count].decode(bytes.replace(/^0x/i,''), offsets[count],  types[count], count);
  // This validation will never occur
  decodedValue = (decodedValue === '0x') ? null : decodedValue; 

  // ...
});
```

My solution for this issue was to move the validation one step back. But before doing a PR, I would like to check with you if what I am doing is correct or the error I am having in the newest version, is an expected behaviour. This is what I did:

```javascript
outputs.forEach(function (output, i) {
  bytes = bytes.replace(/^0x/i,'')
  var decodedValue = bytes ?
    solidityTypes[count].decode(bytes, offsets[count],  types[count], count) : null // If bytes is an empty string set value as null

  // ...
});
```

Thanks!

**Comment by PetroJames** [comment]
I'm getting similar issue i believe:

#1643

**Comment by nivida** [comment]
Will be fixed in the next release (v1.0.0.beta-36)

https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-abi/src/index.js#L225

**Comment by ianaya89** [comment]
@nivida thanks!
